src/coreclr/src/jit/emitarm64.cpp

```diff
@@ -5539,6 +5539,12 @@ void emitter::emitIns_R_R_I(
                 assert(!"Instruction cannot be encoded: IF_LS_2B");
             }
         }
+
+        // Is the ldr/str even necessary?
+        if (emitComp->opts.OptimizationEnabled() && IsRedundantLdStr(ins, reg1, reg2, imm, size, fmt))
```

Comment:
I believe that it's the case that this could potentially be a volatile load or store, but since `IsRedundantLdStr` handles only `ldr` and `str`, those won't be optimized. I think it would be good to add a comment here to that effect.

Reply:
Since `IsRedundantLdStr()` will look for `ldr` and `str` that are next to each other, for volatile load or store (which will have memory barrier instruction between them), the method will return false and we won't optimize it. I will add the comment.